internal/state/statuses.go

```diff
@@ -15,32 +15,26 @@ type ListenerStatuses map[string]ListenerStatus
 // HTTPRouteStatuses holds the statuses of HTTPRoutes where the key is the namespaced name of an HTTPRoute.
 type HTTPRouteStatuses map[types.NamespacedName]HTTPRouteStatus
 
+// GatewayStatuses holds the statuses of Gateways where the key is the namespaced name of a Gateway.
+type GatewayStatuses map[types.NamespacedName]GatewayStatus
```

Comment:
Not sure this type actually gets used much (if at all). I see `map[types.NamespacedName]GatewayStatus` used quite a bit, so I'd either get rid of the type definition, or use it in place of the map.

Reply:
It's mostly defined for documentation purposes. But I will use it in place of the map.